src/filepath.c

```diff
@@ -842,7 +845,22 @@ f_chdir(typval_T *argvars, typval_T *rettv)
 	vim_free(cwd);
     }
 
-    if (curwin->w_localdir != NULL)
+    if (argvars[1].v_type != VAR_UNKNOWN)
+    {
+	char_u *s = tv_get_string(&argvars[1]);
+	if (STRCMP(s, "global") == 0)
+	    scope = CDSCOPE_GLOBAL;
+	else if (STRCMP(s, "window") == 0)
+	    scope = CDSCOPE_WINDOW;
+	else if (STRCMP(s, "tabpage") == 0)
+	    scope = CDSCOPE_TABPAGE;
```

Comment:
The order of the scopes seems a bit inconsistent. The order should be `CDSCOPE_GLOBAL` -> `CDSCOPE_TABPAGE` -> `CDSCOPE_WINDOW` (from outermost to innermost).